modules/user_accounts/ajax/rejectUser.php

```diff
@@ -1,122 +1,63 @@
-    * users with account activity can't be rejected by setting
-    * their status to pending.
-    *
-    * @param string $identifier of account to verify
-    *
-    * @return boolean true if Password_hash is null and
-    * Pending_approval is Yes, false otherwsie
-    *
-    * @throws \Lorisexception if the userID isn't found in the database.
-    */
-function _canRejectAccount($identifier)
-{
-    $DB = Database::singleton();
-    try{
-        $fields =  $DB->pselect(
-            "SELECT Password_hash, Pending_approval
-                       FROM users
-                       WHERE UserID=:id",
-            array("id" => $identifier)
-        );
-        if (is_null($fields[0]['Password_hash'])
-            && $fields[0]['Pending_approval']=="Y"
-        ) {
-            return true;
-        }
-        return false;
-    } catch (DatabaseException $e) {
-        throw new LorisException("User not found in database", USER_NOT_FOUND);
-        exit(1);
-    }
+if (!isset($_POST['identifier'])) {
+    header("HTTP/1.1 400 Bad Request");
+    header("Content-Type: text/plain");
+    exit(
+        "No identifier supplied"
+    );
 }
 
+$rejectee = \User::factory($_POST['identifier']);
+$username = $rejectee->getUsername();
 
-    /**
-    * Send DB query to remove user based on userID, checking that the
-    * user sending the request has admin permissions and the account
-    * can be removed (no password hash => has never had an activity on
-    * Loris, and is pending).
-    *
-    * @return void
-    */
-    // @codingStandardsIgnoreStart
-    function reject_user($identifier)
-    {
-    // @codingStandardsIgnoreEnd
-        $DB      = Database::singleton();
-        $config  = NDB_Config::singleton();
-        $baseURL = $config->getSetting('url');
-
-        $redirect = $baseURL . "/user_accounts/";
+if (empty($username) || $rejectee instanceof AnonymousUser) {
+    header("HTTP/1.1 404 Not Found");
+    header("Content-Type: text/plain");
+    exit(
+        "This account does not exist"
+    );
+}
 
-    if (!_hasPerm()) {
-        throw new LorisException(
-            "You do not have the correct permissions for this 
-            operation (need admin or user accounts)",
-            INCORRECT_PERMISSION
-        );
-        exit(1);
-    } else {
-        if (!_canRejectAccount($identifier)) {
-            throw new LorisException(
-                "This account is active and connot be rejected",
-                ACCOUNT_ACTIVE
-            );
-            exit(1);
-        } else {
-            $DB->delete('users', array("UserID" => $identifier));
-            exit;
-        }
-    }
-    }
+if (!Edit_User::canRejectAccount($rejectee)) {
+    header("HTTP/1.1 403 Forbidden");
+    header("Content-Type: text/plain");
+    exit(
+        "This account is active and cannot be rejected"
+    );
+}
 
+(\Database::singleton())->delete('users', array("UserID" => $username));
```

Comment:
can a not-yet-accepted user have any entries in `user_perm_rel`? if so, delete those too. if not, we good

Reply:
https://github.com/aces/Loris/blob/9cefbcd13844cb44bc4c979eeb9af962676bdbf6/SQL/0000-00-01-Permission.sql#L59

yes, a user pending approval can have entries in `user_perm_rel` but the FK on that table has an on delete cascade clause to it so permissions are automatically deleted along with the entry is `users`